daemon/cmd/kube_proxy_replacement.go

```diff
@@ -434,7 +435,7 @@ func finishKubeProxyReplacementInit(isKubeProxyReplacementStrict bool) {
 		case (option.Config.EnableIPv4Masquerade || option.Config.EnableIPv6Masquerade) &&
 			!option.Config.EnableBPFMasquerade:
 			msg = fmt.Sprintf("BPF host routing requires %s.", option.EnableBPFMasquerade)
-		case !hasHardwareAddress():
+		case !mac.HaveMACAddr(option.Config.Devices):
```

Comment:
If you break apart the commits like mentioned previously, then this commit can be squashed into the second commit, and this delta between commits won't exist.